packages/block-library/src/index.js

```diff
@@ -104,66 +104,82 @@ const registerBlock = ( block ) => {
 	registerBlockType( name, settings );
 };
 
+/**
+ * Function to get all the core blocks in an array.
+ *
+ * @example
+ * ```js
+ * import { __experimentalGetCoreBlocks } from '@wordpress/block-library';
+ *
+ * const coreBlocks = __experimentalGetCoreBlocks();
+ * ```
+ */
+export const __experimentalGetCoreBlocks = () => [
+	// Common blocks are grouped at the top to prioritize their display
+	// in various contexts — like the inserter and auto-complete components.
+	paragraph,
+	image,
+	heading,
+	gallery,
+	list,
+	quote,
+
+	// Register all remaining core blocks.
+	shortcode,
+	archives,
+	audio,
+	button,
+	buttons,
+	calendar,
+	categories,
+	code,
+	columns,
+	column,
+	cover,
+	embed,
+	file,
+	group,
+	window.wp && window.wp.oldEditor ? classic : null, // Only add the classic block in WP Context
+	html,
+	mediaText,
+	latestComments,
+	latestPosts,
+	missing,
+	more,
+	nextpage,
+	preformatted,
+	pullquote,
+	rss,
+	search,
+	separator,
+	reusableBlock,
+	socialLinks,
+	socialLink,
+	spacer,
+	subhead,
+	table,
+	tagCloud,
+	textColumns,
+	verse,
+	video,
+];
+
 /**
  * Function to register core blocks provided by the block editor.
  *
+ * @param {Array} blocks An optional array of the core blocks being registered.
+ *
  * @example
  * ```js
  * import { registerCoreBlocks } from '@wordpress/block-library';
  *
  * registerCoreBlocks();
  * ```
  */
-export const registerCoreBlocks = () => {
-	[
-		// Common blocks are grouped at the top to prioritize their display
-		// in various contexts — like the inserter and auto-complete components.
-		paragraph,
-		image,
-		heading,
-		gallery,
-		list,
-		quote,
-
-		// Register all remaining core blocks.
-		shortcode,
-		archives,
-		audio,
-		button,
-		buttons,
-		calendar,
-		categories,
-		code,
-		columns,
-		column,
-		cover,
-		embed,
-		file,
-		group,
-		window.wp && window.wp.oldEditor ? classic : null, // Only add the classic block in WP Context
-		html,
-		mediaText,
-		latestComments,
-		latestPosts,
-		missing,
-		more,
-		nextpage,
-		preformatted,
-		pullquote,
-		rss,
-		search,
-		separator,
-		reusableBlock,
-		socialLinks,
-		socialLink,
-		spacer,
-		subhead,
-		table,
-		tagCloud,
-		textColumns,
-		verse,
-		video,
-	].forEach( registerBlock );
+export const registerCoreBlocks = (
+	blocks = __experimentalGetCoreBlocks()
+) => {
+	blocks.forEach( registerBlock );
 
 	setDefaultBlockName( paragraph.name );
```

Comment:
I note that there'd be issues if one of these blocks was not passed in the blocks param.

That could be considered using the function incorrectly, but we could also add some guards to prevent errors.